src/main/java/org/broadinstitute/hellbender/engine/ReadInput.java

```diff
@@ -0,0 +1,170 @@
+package org.broadinstitute.hellbender.engine;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Class to represent a Read-containing input file. Tools should declare @Argument-annotated fields of
+ * this type (or Collections of this type), and the engine will automatically discover
+ * them at runtime.
+ *
+ * DO NOT ATTEMPT TO INSTANTIATE THIS CLASS DIRECTLY! ReadInputs must be instantiated by the argument-parsing
+ * system only. This is why the constructor is marked as protected.
```

Comment:
Also, I'm not sure that this warning should apply to `ReadInputs` -- the equivalent warning for `FeatureInput` was put in place due to the `FeatureInput` auto-discovery in `FeatureManager`, but there is no equivalent to this for `ReadInput`